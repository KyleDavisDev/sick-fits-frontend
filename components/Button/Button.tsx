import * as React from "react";
import { StyledButton } from "./ButtonStyles";

interface IButtonProps {
  children: any;
  disabled?: boolean;
  onClick: () => void;
}

const Button: React.FunctionComponent<IButtonProps> = props => {
  return (
    <StyledButton onClick={props.onClick} disabled={props.disabled}>
      {props.children}
    </StyledButton>
  );
};

export default Button;
