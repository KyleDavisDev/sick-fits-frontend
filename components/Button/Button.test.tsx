import * as React from "react";
import * as enzyme from "enzyme";

import Button from "./Button";

describe("<Button>", () => {
  it("renders", () => {
    const wrapper = enzyme.shallow(
      <Button onClick={() => {}}>Click me!</Button>
    );
    expect(wrapper).toBeTruthy();
  });

  it("renders correct text", () => {
    let wrapper = enzyme.shallow(<Button onClick={() => {}}>Text here</Button>);
    expect(
      wrapper
        .find("StyledButton")
        .render()
        .text()
    ).toEqual("Text here");

    // wrapper = enzyme.render(<Button onClick={() => {}}>Other text</Button>);
    // expect(wrapper.find("button").text()).toEqual("Other text");
  });
});
