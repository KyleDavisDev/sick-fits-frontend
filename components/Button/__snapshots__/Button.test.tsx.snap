// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> snapshot 1`] = `
<StyledButton
  onClick={[Function]}
>
  Click me!
</StyledButton>
`;
